spring-cloud-starter-dataflow-server-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/server/cloudfoundry/config/CloudFoundryEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.cloudfoundry.config;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.cloud.dataflow.server.config.features.FeaturesProperties;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+
+/**
+ * An {@link EnvironmentPostProcessor} that sets the original task feature property based on the property
+ * 'spring.cloud.dataflow.features.experimental.tasksEnabled' value.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public class CloudFoundryEnvironmentPostProcessor implements EnvironmentPostProcessor {
+
+	private static final String FEATURES_PREFIX = FeaturesProperties.FEATURES_PREFIX + ".";
+
+	@Override
+	public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
+		Map<String, Object> propertiesToOverride = new HashMap<>();
+		String isTasksEnabled = environment.getProperty(FEATURES_PREFIX + "experimental.tasksEnabled");
+		propertiesToOverride.put(FEATURES_PREFIX + FeaturesProperties.TASKS_ENABLED, Boolean.valueOf(isTasksEnabled));
```

Comment:
I think we should  _log_ a warning here if the existing value of the `spring.cloud.dataflow.features.tasksEnabled` had been set to `true`, since we want users to know they should not set that one (without `experimental`) directly, and especially if we are going to override a `true` to `false` here.